### PR TITLE
Comb sort

### LINKED ISSUE #172: Comb Sort
I would like to add Comb Sort to the algorithms

### PR DESCRIPTION
**Pull request by ColeSeverson**
- [X] Read the [contribution guidelines][contrib-guidelines]
- [x] Added [unit tests][unit-tests] (or unit tests have already been added)
- [x] Updated the [Contents][contents]

Closes #172. 
Closes #185.

[contrib-guidelines]: https://github.com/faheel/Algos/blob/master/CONTRIBUTING.md
[unit-tests]: https://github.com/faheel/Algos/blob/master/C%2B%2B/UNIT_TESTS.md
[contents]: https://github.com/faheel/Algos/tree/master/C%2B%2B#contents

### CODE REVIEW

File: C++/include/sorting/comb_sort.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	----------------
+	O(1)
+*/
+
+#include <iostream>
+#include <vector>
+
+#include "./utils.hpp"
+
+void comb_sort(vector<int>& input, const int order = 1, const bool to_show_state = true){
```
Comment (faheel) [inline C++/include/sorting/comb_sort.hpp]: `to_show_state` should be `false` by default.

File: C++/include/sorting/comb_sort.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	unsigned int gap = input.size();
+	double shrink_factor = 1.3; //suggested as the optimal size
+	bool sorted = false;
+
+	while(!sorted){
+		gap = (gap / shrink_factor); //This should automatically floor bc it is an int
```
Comment (faheel) [inline C++/include/sorting/comb_sort.hpp]: This is simpler:
```c++
gap /= shrink_factor;
```

**Review by faheel — CHANGES_REQUESTED** [review]
@ColeSeverson Thank you for contributing! There are only a few minor things to fix and then it'll be good to merge:
* Re-indent code with 4 spaces per indent level.
* Add a single space between `if`, `for`, `while` etc. and the opening parenthesis `(`. This makes the code easier to read as these statements can be distinguished from function calls more easily.
* Drop the changes in `README.md` that were merged from `master`. It's not required.
* Add an entry for comb sort in the contents (in `C++/README.md`).

**Comment by ColeSeverson** [comment]
Implemented the fixes that you requested. It should be up to spec now.

**Comment by ColeSeverson** [comment]
@faheel Implemented the spacing checks that you wanted added, as well as adding it to the README.md

File: C++/include/sorting/comb_sort.hpp (hunk 2 above)
Comment (faheel) [reply C++/include/sorting/comb_sort.hpp]: This is still pending.

File: C++/include/sorting/comb_sort.hpp (hunk 1 above)
Comment (faheel) [reply C++/include/sorting/comb_sort.hpp]: This is still pending.

File: C++/README.md [hunk 3]
```diff
@@ -38,6 +38,7 @@ Implementation of well-known (and some rare) algorithms, in C++.
   * [Radix sort](source/sorting/radix_sort.cpp)
   * [Selection sort](source/sorting/selection_sort.cpp)
   * [Shell sort](source/sorting/shell_sort.cpp)
+  * [Comb sort](source/include/sorting/comb_sort.hpp)
```
Comment (faheel) [inline C++/README.md]: Please maintain alphabetical sorting in the list.

File: C++/include/sorting/comb_sort.hpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
+	Comb Sort by Cole Severson
+	---------
+	A sorting algorithm desinged to improve upon bubble sort by by eliminating small values        that start near the end of the list.
```
Comment (faheel) [inline C++/include/sorting/comb_sort.hpp]: Looks like a newline got deleted here.

**Review by faheel — CHANGES_REQUESTED** [review]
A few changes that were requested in the previous review weren't made:
* Add a single space between `if`, `for`, `while` etc. and the opening parenthesis `(`. This makes the code easier to read as these statements can be distinguished from function calls more easily.
* Move `comb_sort.hpp` under `C++/include/algorithm/sorting/`.

**Comment by ColeSeverson** [comment]
@faheel Ok, I moved Comb_sort.hpp to include/algorithms/sorting/ and added spacing around the parenthesis on conditionals and loops. Let me know if anything else needs changing!

**Comment by ColeSeverson** [comment]
@faheel Fixed the order in the README.md

**Comment by ColeSeverson** [comment]
@faheel All changes should be fixed

**Comment by alxmjo** [comment]
Assuming the tests that are currently running pass, then this should be good to merge.

**Pull request merged**